File: ExampleApp/app/src/main/java/com/motiv/example/AuthApi.java
package com.motiv.example;

import com.google.gson.*;
import com.google.gson.annotations.*;
import com.google.gson.reflect.*;
import retrofit2.*;
import retrofit2.http.*;

public class AuthApi {

    private static AuthApiApi authApiApi;

    public AuthApi(AuthApiApi authApiApi) {
        this.authApiApi = authApiApi;
    }

    public void getToken(
            final com.motiv.example.OnResponseListener<com.motiv.example.AuthToken>
                    onResponseListener) {
        authApiApi
                .getToken()
                .enqueue(
                        new Callback<com.motiv.example.AuthToken>() {
                            @Override
                            public void onResponse(
                                    Call<com.motiv.example.AuthToken> call,
                                    retrofit2.Response<com.motiv.example.AuthToken> response) {

                                onResponseListener.onSuccess(response.body());
                            }

                            @Override
                            public void onFailure(
                                    Call<com.motiv.example.AuthToken> call, Throwable t) {

                                onResponseListener.onError(new Exception(t.getMessage()));
                            }
                        });
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/AuthApi.kt
 
package com.motiv.example
import androidx.fragment.app.*
import com.google.gson.*
import com.google.gson.annotations.*
import com.google.gson.reflect.*
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*
import retrofit2.*
import retrofit2.http.*

public class AuthApi(val authApiApi: AuthApiApi) {

    fun getToken(onResponseListener: com.motiv.example.OnResponseListener<com.motiv.example.AuthToken>) {
        authApiApi!!.getToken().enqueue(object : Callback<com.motiv.example.AuthToken> {
            override fun onFailure(call: Call<com.motiv.example.AuthToken>?, t: Throwable?) {
                onResponseListener.onError(Exception(t?.message))
            }

            override fun onResponse(call: Call<com.motiv.example.AuthToken>?, response: Response<com.motiv.example.AuthToken>?) {
                onResponseListener.onSuccess(response?.body() as com.motiv.example.AuthToken)
            }
        })
    }
}
